libdrgn/drgn.h.in

```diff
@@ -637,6 +645,32 @@ static inline size_t drgn_type_num_members(struct drgn_type *type)
 	return type->_private.num_members;
 }
 
+/**
+ * Get whether a type has template parameters.
+ */
+static inline bool drgn_type_has_template_parameters(struct drgn_type *type)
```

Comment:
The `drgn_type_has_foo()` helpers should only be based on the kind of type it is. In other words, any struct/class/union type "has" template parameters, even if the template parameters are the empty set. Instead, make this consistent with the other `drgn_type_kind_has_foo()` and `drgn_type_has_foo()` functions.